Automattic/jetpack
# VideoPress: reorganize and polish hooks to sync video Data

# branched off from https://github.com/Automattic/jetpack/pull/27192

This PR reorganizes the hooks' files to synchronize the video data in the block editor context and polishes the process.

Fixes #

#### Changes proposed in this Pull Request:

* VideoPress: reorganize and polish hooks to sync video Data

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






Since it's a refactoring process, I suggest functionality tests:

### Test sync initial video data

* Go to block editor
* Create a Video block
* Open the sidebar
* Confirm the browser performs only one request to get the video data

<img width="884" alt="Screen Shot 2022-11-02 at 05 29 49" src="https://user-images.githubusercontent.com/77539/199478980-08e7aad7-8138-4659-95d8-e1e7e4cbf199.png">

* Change title and/or description
* Confirm the data is synced by taking a look at the async requests

<img width="1150" alt="Screen Shot 2022-11-02 at 05 31 30" src="https://user-images.githubusercontent.com/77539/199479276-048fe384-093f-471b-8bb6-9462bc0e025c.png">

Above, the client performs a POST request to update the video data title, hitting the REST API endpoint.

* Go to the VideoPress dashboard
* Edit the video details via the video page

<img width="348" alt="image" src="https://user-images.githubusercontent.com/77539/199479508-7337812e-d956-4833-a63b-b6027c54c44d.png">

<img width="893" alt="Screen Shot 2022-11-02 at 05 34 24" src="https://user-images.githubusercontent.com/77539/199479769-050d8041-865a-41d0-a1de-fc16dcb71783.png">

* Go back to block editor
* hard-refresh
* Confirm the data is synced 
<img width="491" alt="Screen Shot 2022-11-02 at 05 35 22" src="https://user-images.githubusercontent.com/77539/199479944-20234572-289d-459f-a2e5-100fb456357f.png">

### Test sync chapters

* Change video chapters from the description text area
* Confirm the whole flow

https://user-images.githubusercontent.com/77539/199480684-22c40346-09e2-4529-a795-802ed5b7ef94.mov
